services/src/main/java/org/apache/druid/cli/PullDependencies.java

```diff
@@ -83,6 +83,9 @@
                   .put("commons-beanutils", "commons-beanutils")
                   .put("org.apache.commons", "commons-compress")
                   .put("org.apache.zookeeper", "zookeeper")
+                  .put("com.fasterxml.jackson.core", "jackson-databind")
+                  .put("com.fasterxml.jackson.core", "jackson-core")
+                  .put("com.fasterxml.jackson.core", "jackson-annotations")
```

Comment:
I assume this is the part of the PR that fixes

```
When druid.extensions.useExtensionClassloaderFirst is set to true, services fail to start due to duplicate jackson-databind dependencies in extensions.
```
Can you explain how? I don't understand PullDependencies well enough

Reply:
I will try to add as much context as possible - 
Usually, custom classloaders in Java have parent first behavior. That is, say there is a classloader X of which Y is a parent. Now when X is trying to load a class, it will first ask Y to load it if the class is not loaded. When `useExtensionClassloaderFirst` is set to false, we inverse the relationship. If a class is not loaded, X will load this class by itself first. X will delegate loading to Y if it can't find the class. 

Now, this child first behavior can cause a situation where X loads a class SimpleModule and its superclass Module on its own, and the same classes are also loaded by Y too. When you cast a class `SimpleModule` loaded by X to a class `Module` loaded by Y, you will get an exception since classloaders are different for these classes. This is exactly what happens when the broker is loading the extension modules. 

So ideally we want to avoid a situation where you have the same classes loaded by both classloaders. `PROVIDED_CORE` is a way within PullDependencies to exclude certain dependencies being shipped with the extension since they are already being provided by the core. That way, class can only be loaded by the parent classloader.

Reply:
We tested `ITHadoopIndexTest`.